.vib/metallb/ginkgo/metallb_test.go

```diff
@@ -0,0 +1,74 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("MetalLB", func() {
+	var typedClient corev1.CoreV1Interface
+	var dynamicClient dynamic.Interface
+	var ctx context.Context
+
+	BeforeEach(func() {
+		typedClient = corev1.NewForConfigOrDie(clusterConfigOrDie())
+		dynamicClient = dynamic.NewForConfigOrDie(clusterConfigOrDie())
+		ctx = context.Background()
+	})
+
+	Context("configured via IPAddressPool & L2Advertisement resources (CRDs)", func() {
+		var ipPoolConfig = IpAddressPoolConfig{
+			name:        "ip-address-pool-vib-tests",
+			addressFrom: "192.168.100.0",
+			addressTo:   "192.168.100.255",
+		}
+		var advertConfig = L2AdvertisementConfig{
+			name:              "l2-advertisement-vib-tests",
+			ipAddressPoolName: ipPoolConfig.name,
+		}
+
+		BeforeEach(func() {
+			createIpAddressPoolResourceOrDie(ctx, dynamicClient, &ipPoolConfig)
+			createL2AdvertisementResourceOrDie(ctx, dynamicClient, &advertConfig)
+		})
+		AfterEach(func() {
+			// Not need to panic here if failed, the cluster is expected to clean up with the undeployment
+			dynamicClient.Resource(l2AdvertisementType).Namespace(*namespace).Delete(ctx, advertConfig.name, metav1.DeleteOptions{})
+			dynamicClient.Resource(ipAddressPoolType).Namespace(*namespace).Delete(ctx, ipPoolConfig.name, metav1.DeleteOptions{})
+		})
+
+		When("a Service resource is created", func() {
+			var service *v1.Service
+			var serviceName string = "lb-vib-tests"
+
+			BeforeEach(func() {
+				var err error
+				createServiceOrDie(ctx, typedClient, serviceName)
+				service, err = typedClient.Services(*namespace).Get(ctx, serviceName, metav1.GetOptions{})
```

Comment:
This `get()` can be avoided, as the `create()` returns the created resource.

Reply:
I have tried to update the logic as you suggested, but the `create()` returns the newly created object, right before MetalLB has updated its configuration (and assigned an IP to the service).

Using the `get()` to query the API returns the service once MetalLB has managed to noticed it, though.